sre-recipes/recipes/currencies_recipe/currencies_recipe.py

```diff
@@ -48,103 +41,74 @@ def _deploy_state(state):
-        logging.info('Cluster has been authenticated')
+        Recipe._run_command(delete_pod_command)
 
     def break_service(self):
         """
         Rolls back the working version of the given service and deploys the
         broken version of the given service
         """
-        print("Deploying broken service...")
-        self._auth_cluster()
+        print('Deploying broken service...')
+        Recipe._auth_cluster()
         self._deploy_state(True)
-        print("Done")
+        print('Done')
         logging.info('Deployed broken service')
 
     def restore_service(self):
         """
         Rolls back the broken version of the given service and deploys the
         working version of the given service
         """
-        print("Deploying working service...")
-        self._auth_cluster()
+        print('Deploying working service...')
+        Recipe._auth_cluster()
         self._deploy_state(False)
-        print("Done")
+        print('Done')
         logging.info('Deployed working service')
 
-    def _service_multiple_choice(self):
+    def hint(self):
+        """
+        Provides a hint about finding the root cause of this recipe
+        """
+        print('Giving hint for recipe')
+        external_ip_command = "kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}'"
+        ip, error = Recipe._run_command(external_ip_command)
+        ip = ip.decode("utf-8").replace("'", '')
+        print('Visit the external IP of the demo application to see if there are any visible changes: http://{}'.format(ip))
+        get_project_command = "gcloud config list --format value(core.project)"
+        project_id, error = Recipe._run_command(get_project_command)
+        project_id = project_id.decode("utf-8").replace('"', '')
+        print('Use Monitoring Dashboards to see metrics associated with each service: https://console.cloud.google.com/monitoring/dashboards?project={}'.format(project_id))
+
+    def verify_broken_service(self):
         """
         Displays a multiple choice quiz to the user about which service
         broke and prompts the user for an answer
         """
-        print("Which service broke?\n"
-                "\t[a] ad service \n"
-                "\t[b] cart service \n"
-                "\t[c] checkout service \n"
-                "\t[d] currency service \n"
-                "\t[e] email service \n"
-                "\t[f] frontend service \n"
-                "\t[g] payment service \n"
-                "\t[h] product catalog service \n"
-                "\t[i] recommendation service \n"
-                "\t[j] redis \n"
-                "\t[k] shipping service")
-        answer = input('Your answer: ')
-        return answer
+        prompt = 'Which service has a breakage?'
+        choices = ['email service', 'frontend service', 'ad service', 'recommendation service', 'currency service']
+        answer = 'frontend service'
```

Comment:
Re-typing 'frontend service' in the choices and answer seems a little risky to me. Strings should really only be defined once in code. What if it changes in one place but not the other?

I already suggested using an index for the answer before. Another option might be to make choices a list of tuples containing the name and the success value: `[('email service', False), ('frontend service', True) ...]`

This way would keep things more tightly coupled. But this is just a suggestion, I don't think it's a huge issue as is either

Reply:
Hmm I do agree it makes it dangerous to have the string declared twice - I think using an index might be confusing to the developer because of the 0 vs 1 start which I guess we could just start the index at 0 regardless? I think the tuple also makes it more work for the developer since they have to specifically designate whether each option is correct or incorrect. I'm not too sure what the best option here is- I think we want to keep it relatively simple for people to create these multiple choice questions but also avoid room for errors.

Reply:
Yeah, I think it's probably fine as is. If we were expecting this class to grow a lot in complexity we might want to re-think it, but I'm not too worried for our use case. Especially since you added a check at the top of the function